saleor/core/taxes/avatax/__init__.py

```diff
@@ -0,0 +1,388 @@
+import json
+import logging
+from datetime import date
+from typing import TYPE_CHECKING, Any, Dict, List, Optional, Union
+from urllib.parse import urljoin
+
+import requests
+from django.conf import settings
+from django.core.cache import cache
+from django.utils.translation import pgettext_lazy
+from requests.auth import HTTPBasicAuth
+
+from saleor.core.utils import get_company_address
+
+from .. import charge_taxes_on_shipping, include_taxes_in_prices
+
+if TYPE_CHECKING:
+    from ....checkout.models import Checkout
+    from ....order.models import Order
+
+META_FIELD = "avatax"
+
+logger = logging.getLogger(__name__)
+
+
+class TransactionType:
+    INVOICE = "SalesInvoice"
+    ORDER = "SalesOrder"
+
+
+class CustomerErrors:
+    DEFAULT_MSG = pgettext_lazy(
+        "We are not able to calculate taxes for your order. Please try later"
+    )
+    ERRORS = ("InvalidPostalCode", "InvalidAddress", "MissingAddress")
+
+    @classmethod
+    def get_error_msg(cls, error: dict) -> str:
+        error_code = error.get("code")
+        if error_code in cls.ERRORS:
+            return error.get("message", cls.DEFAULT_MSG)
+        return cls.DEFAULT_MSG
+
+
+def get_api_url() -> str:
+    """Based on settings return sanbox or production url"""
+    if settings.AVATAX_USE_SANDBOX:
+        return "https://sandbox-rest.avatax.com/api/v2/"
+    return "https://rest.avatax.com/api/v2/"
+
+
+def api_post_request(
+    url: str,
+    data: Dict[str, Any],
+    username: str = settings.AVATAX_USERNAME_OR_ACCOUNT,
+    password: str = settings.AVATAX_PASSWORD_OR_LICENSE,
+) -> Dict[str, Any]:
+    try:
+        auth = HTTPBasicAuth(username, password)
+        response = requests.post(url, auth=auth, data=json.dumps(data), timeout=2)
```

Comment:
We might want to not hard-cord the timeout. Users should be able to customize that.

Is 2 really enough? On high load on the workers, we might hit a few timeouts (might)

Reply:
Ideally, our own load would not affect the timeout as it's the limit placed upon the external resource. We should definitely set a timeout but I agree that it may need to be configurable. I think 10 seconds is reasonable as a default value unless it's part of a series of requests and they add up to several minutes in total.